test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -88,3 +88,25 @@ Describe "Import-Module for Binary Modules in GAC" -Tags 'CI' {
         }
     }
 }
+
+Describe "Import-Module for Binary Modules" -Tags 'CI' {
+
+    It "PS should try to load the assembly from file path first" {
+ $src = @"
+using System;
+namespace ModuleCmdlets
+{ 
+  public class TestBinaryModuleCmdlet1Command

```

Comment:
You should make this a complete cmdlet that can be called. That is the real scenario that needs to be tested.